tests/acceptance/support-test.js

```diff
@@ -205,6 +205,8 @@ test detail
       };
 
       await visit('/crates/nanomsg');
+      assert.strictEqual(currentURL(), '/crates/nanomsg');
+      assert.dom('[data-test-id="crate-sidebar"]').exists();
```

Comment:
how would this fix it?

Reply:
Ah, wait, sorry. I should have used `waitFor` rather than `assert.dom` 🤦.

Reply:
I just changed it to wait for the button to exist.

Reply:
> I should have used `waitFor` rather than `assert.dom` 🤦.

I guess that should work, though I'm wondering why it is needed here. iirc the `link-crate-report` button should be rendered directly and there are no network requests or other async things that would prevent that. feels like something else might be wrong.

ultimately, the assertion architecture of playwright where `waitFor()` is the default is probably better than the test waiter architecture of Ember.js 😅

Reply:
> irc the `link-crate-report` button should be rendered directly and there are no network requests or other async things that would prevent that. . feels like something else might be wrong.

Yeah, I have no clue what's going on here either! It only depends on the crate (@crate.name), which should be resolved in the routes, but it's failing somehow for no obvious reason. This led me to wait for it explicitly as a naive solution. I should investigate the root cause in more detail in the future.



> ultimately, the assertion architecture of playwright where `waitFor()` is the default is probably better than the test waiter architecture of Ember.js 😅

Indeed, the auto-retry is much more ergonomic.